app/views/root/_gem_base.html.erb

```diff
@@ -16,15 +20,17 @@
       short_description: @emergency_banner.short_description,
     }
   end
-
-  user_satisfaction_survey = '<div id="user-satisfaction-survey-container"></div>'
-  global_bar = render "components/global_bar"
+  global_bar = ''
+  user_satisfaction_survey = '<div id="user-satisfaction-survey-container"></div>' unless omit_user_satisfaction_survey
+  global_banner = render "components/global_bar" unless omit_global_banner
+  global_bar << user_satisfaction_survey if user_satisfaction_survey
+  global_bar << global_banner if global_banner
 %>
 
 <%= render "govuk_publishing_components/components/layout_for_public", {
   emergency_banner: emergency_banner.presence,
   full_width: full_width,
-  global_bar: user_satisfaction_survey + global_bar,
+  global_bar: !global_bar.empty? ? global_bar  : nil,
```

Comment:
Could this just be `global_bar: global_bar`? You'd get an empty string rather than nil, I don't know if that would cause a problem.

Reply:
I think [`presence`](https://apidock.com/rails/Object/presence) might do the trick here - it's equivalent to `!object.empty? ? object : nil`:

```suggestion
  global_bar: global_bar.presence
```

Reply:
thanks both, seems like `global_bar: global_bar` does the job just fine. 😄